Spotti/Spotti/View Controllers/AddingFriendsViewController.m

```diff
@@ -166,23 +177,30 @@ - (void)mapSearch{
                 NSLog(@"%@",error.localizedDescription);
             }
             NSMutableDictionary *prios = [[NSMutableDictionary alloc] init];
-                for(MKMapItem *item in response.mapItems){
-                    float lat = item.placemark.coordinate.latitude;
-                    float longit = item.placemark.coordinate.longitude;
-                    for(GymUser *user in self.allUsers){
-                        CLLocationDistance distance = [[[CLLocation alloc]initWithLatitude:lat longitude:longit] distanceFromLocation:[[CLLocation alloc]initWithLatitude:user.lastLocation.latitude longitude:user.lastLocation.longitude]];
-                        NSString *key = user.username;
-                        if(![[prios allKeys] containsObject:key]){
-                            [prios setObject:[NSNumber numberWithInt:1] forKey:key];
-                        }
-                        if(distance > 1610){
-                            prios[key] = [NSNumber numberWithInt:[prios[key] intValue] + 1];
+            for(MKMapItem *item in response.mapItems){
+                float lat = item.placemark.coordinate.latitude;
+                float longit = item.placemark.coordinate.longitude;
+                for(GymUser *user in self.allUsers){
+                    CLLocationDistance distance = [[[CLLocation alloc]initWithLatitude:lat longitude:longit] distanceFromLocation:[[CLLocation alloc]initWithLatitude:user.lastLocation.latitude longitude:user.lastLocation.longitude]];
+                    NSString *key = user.username;
+                    if(![[prios allKeys] containsObject:key]){
+                        NSInteger friendsInCommon = [self friendsInCommon:user];
+                        if([item.name isEqualToString:[GymUser currentUser].gym] && [user.gym isEqualToString:[GymUser currentUser].gym]){
+                            [prios setObject:[NSNumber numberWithInt:friendsInCommon * 1.5] forKey:key];
                         }
-                        if([item.name isEqualToString:[GymUser currentUser].gym] && ![user.gym isEqualToString:[GymUser currentUser].gym]){
-                            prios[key] = [NSNumber numberWithInt:[prios[key] intValue] + 1];
+                        else {
+                            [prios setObject:[NSNumber numberWithInt:friendsInCommon] forKey:key];
                         }
                     }
+                    if(distance <= 1610){
+                        NSInteger prio = [item.name isEqualToString:[GymUser currentUser].gym] && [user.gym isEqualToString:[GymUser currentUser].gym] ? 3 : 1;
+                        prios[key] = [NSNumber numberWithInt:[prios[key] intValue] + prio];
```

Comment:
Do you mind adding your priority calculation logic into the PR description?

Reply:
i.e. only calculate distance within 1610 miles; common friends has weight 1.5; ...